src/main/kotlin/se/uulm/snowballr/backend/service/ProjectService.kt

```diff
@@ -196,6 +202,50 @@ class ProjectService(
         )
     }
 
+    /**
+     * Retrieves the gRPC representation of a project's paper, including associated metadata such as authors,
+     * backward references, and reviews. Ensures that the current user has the necessary permissions to access
+     * the project's paper. Throws an exception if the user lacks access rights.
+     *
+     * @param projectPaper The domain object representing the project paper, containing its metadata and identifiers.
+     * @param projectId The unique identifier of the project to which the paper belongs.
+     * @return The gRPC representation of the project's paper, including associated metadata.
+     * @throws UnauthorizedException.Single If the current user does not have the required read access to the project.
+     */
```

Comment:
```suggestion
    /**
     * Retrieves the gRPC representation of a project's paper, including associated data such as authors,
     * backward references, and reviews. Furthermore, it ensures that the current user has the necessary permissions to access
     * the project's paper.
     *
     * @param projectPaper The project paper, containing the ids linking to the associated data.
     * @param projectId The unique identifier of the project to which the paper belongs.
     * @return The gRPC representation of the project's paper, including associated data.
     * @throws UnauthorizedException.Single If the current user does not have the required read access to the project.
     */
```

Please check your comments are generating them (and this looks like a generated comment)

Reply:
Good point. Sorry, I should re-read them after they are generated.. You are right.